nnpdf_data/nnpdf_data/commondata/ATLAS_WJ_8TEV/uncertainties_WP-PT.yaml

```diff
@@ -0,0 +1,1402 @@
+    treatment: ADD
+    type: CORR
+  art_stat_corr_9:
+    description: Artificial statistical uncertainty 9
+    treatment: ADD
+    type: CORR
+  art_stat_corr_10:
+    description: Artificial statistical uncertainty 10
+    treatment: ADD
+    type: CORR
+  art_stat_corr_11:
+    description: Artificial statistical uncertainty 11
+    treatment: ADD
+    type: CORR
+  art_stat_corr_12:
+    description: Artificial statistical uncertainty 12
+    treatment: ADD
+    type: CORR
+  art_stat_corr_13:
+    description: Artificial statistical uncertainty 13
+    treatment: ADD
+    type: CORR
+  art_stat_corr_14:
+    description: Artificial statistical uncertainty 14
+    treatment: ADD
+    type: CORR
+  art_stat_corr_15:
+    description: Artificial statistical uncertainty 15
+    treatment: ADD
+    type: CORR
+  art_stat_corr_16:
+    description: Artificial statistical uncertainty 16
+    treatment: ADD
+    type: CORR
+  JetScaleEff1:
+    description: 'Systematic: JetScaleEff1'
+    treatment: MULT
+    type: SYSATLASW1
+  JetScaleEff2:
+    description: 'Systematic: JetScaleEff2'
+    treatment: MULT
+    type: SYSATLASW2
+  JetScaleEff3:
+    description: 'Systematic: JetScaleEff3'
+    treatment: MULT
+    type: SYSATLASW3
+  JetScaleEff4:
+    description: 'Systematic: JetScaleEff4'
+    treatment: MULT
+    type: SYSATLASW4
+  JetScaleEff5:
+    description: 'Systematic: JetScaleEff5'
+    treatment: MULT
+    type: SYSATLASW5
+  JetScaleEff6:
+    description: 'Systematic: JetScaleEff6'
+    treatment: MULT
+    type: SYSATLASW6
+  JetScaleEta1:
+    description: 'Systematic: JetScaleEta1'
+    treatment: MULT
+    type: SYSATLASW7
+  JetScaleEta2:
+    description: 'Systematic: JetScaleEta2'
+    treatment: MULT
+    type: SYSATLASW8
+  JetScaleHighPt:
+    description: 'Systematic: JetScaleHighPt'
+    treatment: MULT
+    type: SYSATLASW9
+  JetScaleMC:
+    description: 'Systematic: JetScaleMC'
+    treatment: MULT
+    type: SYSATLASW10
+  JetScalePileup1:
+    description: 'Systematic: JetScalePileup1'
+    treatment: MULT
+    type: SYSATLASW11
+  JetScalePileup2:
+    description: 'Systematic: JetScalePileup2'
+    treatment: MULT
+    type: SYSATLASW12
+  JetScalePileup3:
+    description: 'Systematic: JetScalePileup3'
+    treatment: MULT
+    type: SYSATLASW13
+  JetScalePileup4:
+    description: 'Systematic: JetScalePileup4'
+    treatment: MULT
+    type: SYSATLASW14
+  JetScaleFlav1Known:
+    description: 'Systematic: JetScaleFlav1Known'
+    treatment: MULT
+    type: SYSATLASW15
+  JetScaleFlav2:
+    description: 'Systematic: JetScaleFlav2'
+    treatment: MULT
+    type: SYSATLASW16
+  JetScaleBjet:
+    description: 'Systematic: JetScaleBjet'
+    treatment: MULT
+    type: SYSATLASW17
+  JetScalepunchT:
+    description: 'Systematic: JetScalepunchT'
+    treatment: MULT
+    type: SYSATLASW18
+  JetResolution10:
+    description: 'Systematic: JetResolution10'
+    treatment: MULT
+    type: SYSATLASW19
```

Comment:
I haven't looked at _every_ uncertainty, but if they are to be correlated between experiments, they should have the same name. And that means they probably should keep the same name as in the legacy files.

Reply:
There are two types of correlation to take into account. The first one is the correlation intra-set, namely between the $W^+$ and $W^-$ productions within this same dataset. In this case, the important thing is that the same name is used between the uncertainties files for $W^+$ and $W^-$ productions, respectively. In the legacy version was used `ATLASWX`,  while now I'm using `SYSATLASWX` because I like it more. But the name does not make any difference as long as it is the same between the two productions. (also, I explicitly checked that `ATLASWX` is only used within the legacy implementation of this dataset).

The other type of correlation is inter-experiment, namely between different experiments. The experiments that have to be correlated are shown in the table that I have included in the description. In this case, and for a subset of systematic uncs.,  the same name must be used between different experiments. However, accounting for these inter-experiment correlations requires coordination between the different re-implementations affected by these inter-correlations.

As I said, I would leave the inter-correlations for later, and the 'default' variant will only account for the first type of correlations, namely the one between  $W^+$ and $W^-$ productions. 

Is it now clear?

Reply:
> . In the legacy version was used ATLASWX, while now I'm using SYSATLASWX because I like it more.
> (also, I explicitly checked that ATLASWX is only used within the legacy implementation of this dataset).

So, one thing is that the name in the old one was `ATLASWJ<X>` it should keep the J `SYSATLASWJ<X>`, but aside from that, if you really want to do this you need to update also the legacy uncertainty files, otherwise one won't be able to compute the chi2 using e.g., the legacy variant for W- and the new one for W+.

All in all, I would suggest keeping the old names as it keeps things simpler. Once all datasets have been ported we can systematically change all names, the least moving parts we have the best.

_if_ (and only _if_) the resulting (t0) covariance matrix is exactly the same for the old and new implementation, then you can just remove the old file and keep the new ones (and make the legacy variant point to the new files). Then the change of name wouldn't matter.

Reply:
> So, one thing is that the name in the old one was ATLASWJ<X> it should keep the J SYSATLASWJ<X>, but aside from that, if you really want to do this you need to update also the legacy uncertainty files, otherwise one won't be able to compute the chi2 using e.g., the legacy variant for W- and the new one for W+.

Ok, I agree that if one were to compute the chi2 with this combination, then the two names should be the same.  But I would highly discourage this kind of mixing because some of the uncertainties have changed wrt the legacy implementation, and correlating the new implementation with the legacy one is wrong IMHO. But if you wish, I can use the same name for both legacy and new implementations. What do you think?

Regarding the inter-correlations, we need to choose a set of names that will be used across different experiments. As far as I know, these correlations were not taken into account in NNPDF4.0. However, this data set comes with the legacy variant `sys_ATLAS` which uses different names for those uncertainties that should be correlated between experiments (e.g. `JET_FLAV_RESP_1`).  I haven't seen these names in any other dataset in the common data, meaning that these uncertainties cannot be correlated across experiments even if one chooses the variant`....._sys_ATLAS.yaml`. Is that correct?

Reply:
At the moment we don't correlate betwen different experiments at all, once we start doing it in the future we will have to revisit many datasets to make sure we didn't make any mistakes anyway...

 > But if you wish, I can use the same name for both legacy and new implementations. What do you think?

As I said, my suggestion is to keep the names the same as in the old implementation to keep things simpler and make comparison easier, but if you feel really strongly about the change of name go ahead.
The only thing is the `J` is necessary here.

> I would highly discourage this kind of mixing

Sure, but there is no need to add a source of mistakes that would be very hard to spot.